### PR TITLE
feat: add custom dimensons support

### PR DESCRIPTION
**Pull request by jozan**

### CODE REVIEW

File: android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java [hunk 1]
```diff
@@ -24,6 +28,8 @@
 
   private final ReactApplicationContext reactContext;
   private Tracker tracker;
+  private DimensionQueue mOneTimeDimensionQueue;
```
Comment (aeirola) [inline android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java]: I'm not sure how safe it is to reuse these dimension queues. Looking at the source https://github.com/matomo-org/matomo-sdk-android/blob/v4.1.2/tracker/src/main/java/org/matomo/sdk/extra/DimensionQueue.java#L42 it seems as if elements are removed from the queue only in cases where they don't rewrite existing values. This could lead to strange bugs where some values custom dimension values would be present in future events even though a user would have attempted to clear the value.

File: android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java [hunk 2]
```diff
@@ -24,6 +28,8 @@
...
+  private List<CustomDimension> mCustomDimensions = new ArrayList<>();
```
Comment (aeirola) [inline android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java]: Does the order of the dimensions matter? Would be it be easier to use a map so that additions and removals do not need to iterate over the whole data structure?

File: android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java [hunk 3]
```diff
@@ -77,11 +86,45 @@ public void trackEvent(String category, String action, ReadableMap optionalParam
...
   }
-}
\ No newline at end of file
+
+  @ReactMethod
+  @SuppressWarnings("unused")
+  public void setCustomDimension(int dimensionId, @Nullable String value, Promise promise) {
+    try {
+      if (value != null) {
+        mCustomDimensions.add(new CustomDimension(dimensionId, value));
```
Comment (aeirola) [inline android/src/main/java/com/terveystalo/react_native/matomo_sdk/RNMatomoSdkModule.java]: What happens if a value with the given id already exists? Then there are two, possibly different, values in the dimensions. Which one will be sent to the server?

**Review by aeirola — CHANGES_REQUESTED** [review]
In general, the code looks good, but I think there are some potential bugs in the handling of the custom dimension on the android side. I think it would be good the check the edge cases and fix as necessary.

**Review by aeirola — APPROVED** [review]
Looks good!

File: .travis.yml [hunk 4]
```diff
@@ -26,7 +26,7 @@ jobs:
       osx_image: xcode11
       script:
         - npm run test:example
-        - npm run test:ios
+        - travis_wait npm run test:ios
```
Comment (jozan) [inline .travis.yml:29]: Sometimes installing pods takes a bit over 10 minutes to fetch repos without any output. Travis thinks that build is stuck and stops it. `travis_wait` fixes this by outputting something to stdout every minute for 20 minutes or until script outputs something itself. Unfortunately it also hides output until the script returns exit code.

**Comment by aeirola** [comment]
This feature is now released in version 0.4.0!
